full-moon/src/ast/parsers.rs

```diff
@@ -1121,6 +1255,44 @@ fn expect_while_stmt(
     })
 }
 
+#[cfg(feature = "roblox")]
+fn expect_type_declaration(
+    state: &mut ParserState,
+    type_token: TokenReference,
+) -> Result<ast::TypeDeclaration, ()> {
+    let base = match state.current()? {
+        token if token.token_kind() == TokenKind::Identifier => state.consume().unwrap(),
+        token => {
+            state.token_error(token.clone(), "expected type name");
+            // rewrite todo: maybe we can add an error name here to continue parsing?
```

Comment:
Probably not worth it, the end result from an errors perspective would be the same

Reply:
I think luau might be able to recover here, but I will leave it for now. If it seems useful later we can add it